runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -4508,6 +4512,25 @@ void J9::X86::TreeEvaluator::transactionalMemoryJITMonitorEntry(TR::Node
       cg->stopUsingRegister(counterReg);
    }
 
+void
+J9::X86::TreeEvaluator::generateCheckForValueTypeMonitorEnterOrExit(
+      TR::Node *node,
+      TR::LabelSymbol *snippetLabel,
+      TR::CodeGenerator *cg)
+   {
+   if (cg->isMonitorValueType(node) != TR_maybe)
+      return;
+   TR::Register *objectReg = cg->evaluate(node->getFirstChild());
+   auto j9classReg = cg->allocateRegister();
```

Comment:
Can you make these auto types explicit where reasonable eg TR:Register TR_J9VMBase etc

Reply:
done. Leo was suggesting using auto for the TR_J9VMBase case since the type is obvious from the checkcast.